pkg/apiserver/authentication/token/jwt.go

```diff
@@ -45,22 +46,42 @@ type jwtTokenIssuer struct {
 	maximumClockSkew time.Duration
 }
 
-func (s *jwtTokenIssuer) VerifyWithoutClaimsValidation(tokenString string) (user.Info, TokenType, error) {
+func (s *jwtTokenIssuer) VerifyWithoutClaimsValidation(tokenString string) (userInfo user.Info, tokenType TokenType, err error) {
 	parser := &jwt.Parser{
 		SkipClaimsValidation: true,
 	}
-	token, err := parser.Parse(tokenString, s.keyFunc)
-	if err != nil {
-		if token != nil {
-			mapClaims := token.Claims.(jwt.MapClaims)
 
-			return &user.DefaultInfo{
-				Name: mapClaims["sub"].(string),
-			}, "clm.TokenType", nil
+	// set up the default values
+	tokenType = "clm.TokenType"
+	userInfo = &user.DefaultInfo{}
+
+	var token *jwt.Token
+	// TODO consider to verify the token parse result
+	if token, _ = parser.Parse(tokenString, s.keyFunc); token != nil {
+		mapClaims := token.Claims.(jwt.MapClaims)
```

Comment:
We should verify if the type convention is successful

Reply:
Sure.

Reply:
sure. please check https://github.com/kubesphere-sigs/ks-devops/pull/56/commits/8e702d811c2bcde6ac197638d5322dfbf370fb55